specification/iotsecurity/resource-manager/Microsoft.IoTSecurity/preview/2021-02-01-preview/onPremiseSensors.json

```diff
@@ -0,0 +1,352 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "Azure Defender for IoT",
+    "description": "API spec for Microsoft.IoTSecurity (Azure Defender for IoT) resource provider",
+    "version": "2021-02-01-preview"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.IoTSecurity/onPremiseSensors": {
+      "get": {
+        "x-ms-examples": {
+          "List on-premise IoT sensors": {
+            "$ref": "./examples/OnPremiseSensors/List.json"
+          }
+        },
+        "tags": [
+          "On-Premise Sensors"
+        ],
+        "description": "List on-premise IoT sensors",
+        "operationId": "OnPremiseSensors_List",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OnPremiseSensorsList"
+            }
```

Comment:
It's highly unlikely since this resource represent physical, on-premise deployments of our sensor.
AFAIK the current customers don't come anywhere near the standard page limit (1000), so it's pretty safe to assume that we won't need paging ever here. Worst case scenario - support could be added later on if a real need would arise.

Reply:
support added in the future would mean a breaking change.  Since this isn a preview, that should be OK, but expect to be required to change this before the api becomes stable.